python/nano/src/bigdl/nano/pytorch/optim/sparseadam.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""SparseAdam optimizer pytorch implementation."""
+
+# This file is adapted from Lazy Adam optimizer (PyTorch).
+# https://github.com/davda54/lazy-adam/blob/master/lazy_adam.py
+
+# Copyright (c) 2020 David Samuel.
```

Comment:
You should use the original MIT license.

You can just simply copy everything in this file
https://github.com/davda54/lazy-adam/blob/master/LICENSE

Reply:
Ok.